src/ml/neural_net/image_augmentation.hpp

```diff
@@ -237,6 +244,8 @@ class image_augmenter {
      * each inner vector should have the same length as the corresponding input
      * image's annotations vector. */
     std::vector<std::vector<image_annotation>> annotations_batch;
+
+    std::vector<std::pair<float, float>> img_dimensions_batch;
```

Comment:
I’d rather not add this state here. It confuses the purpose of this type, which contains resized/ augmented images. The original image dimensions are not relevant here. 

We need to keep these types streamlined to optimize for future maintainability of the core image augmentation pipeline

Reply:
good!